kong/router.lua

```diff
@@ -1634,7 +1646,8 @@ function _M.new(routes, cache, cache_neg)
     sni = sni or ""
 
     local req_category = 0x00
-    local hits = {}
+
+    clear_tab(hits)
```

Comment:
Note that `clear_tab` does remove the item elements, but it does not change the table size. Since hits "never gets collected" this might result in a "leak" (hits will always increase in size, never decrease). Unless I am reading the docs incorrectly.

Reply:
it's true that `table.clear()` doesn't change the table size; but GC isn't disabled or delayed in any way.   it just avoids replacing the table itself, thus reducing the amount of garbage generated (and doesn't invalidate other references to the same table)